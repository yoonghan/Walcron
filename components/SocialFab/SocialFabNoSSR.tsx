
`use strict`

import * as React from "react";
import Modal from "../Modal";
import EmailSender from "../LetterBox/EmailSender";

export interface SocialFabNoSSRProps {
}

const SocialFabNoSSR: React.SFC<SocialFabNoSSRProps> = ({}) => {
  const [isEmailShown, setEmailShown] = React.useState(false);

  function onOpenEmail() {
    setEmailShown(true);
  }

  function closeCallback() {
    setEmailShown(false);
  }

  return (
    <div className="fab">
      <span className="fab-action-button">
          <i className="fab-action-button__icon"></i>
      </span>
      <ul className="fab-buttons">
        <li className="fab-buttons__item">
          <a className="fab-buttons__link" target="onnew" href="https://www.facebook.com/walcron.coorperation">
            <i className="icon-material icon-material_fb"></i>
          </a>
        </li>
        <li className="fab-buttons__item">
          <a className="fab-buttons__link" onClick={onOpenEmail}>
            <i className="icon-material icon-material_gm"></i>
          </a>
        </li>
      </ul>
      {
        isEmailShown && <Modal cancelCallback={closeCallback}><EmailSender writeTo={"Walcron Website"}/></Modal>
      }
      <style jsx>
        {`
          a:hover {
            cursor: pointer;
          }

          .fab {
            position: fixed;
            width: 56px;
            right: 15px;
            bottom: 15px;
            margin-left: -28px;
          }

          .fab:hover .fab-buttons {
            opacity: 1;
            visibility: visible;
          }

          .fab:hover .fab-buttons__link {
            transform: scaleY(1) scaleX(1) translateY(-16px) translateX(0px);
          }

          .fab-action-button:hover + .fab-buttons .fab-buttons__link:before {
            visibility: visible;
            opacity: 1;
            transform: scale(1);
            transform-origin: right center 0;
            transition-delay: 0.3s;
          }

          .fab-action-button {
            position: absolute;
            bottom: 0;
            display: block;
            width: 56px;
            height: 56px;
            background-color: #000;
            border-radius: 50%;
            box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 1px 5px 0 rgba(0, 0, 0, 0.12), 0 3px 1px -2px rgba(0, 0, 0, 0.2);
          }

          .fab-buttons {
            position: absolute;
            left: 0;
            right: 0;
            bottom: 30px;
            list-style: none;
            margin: 0;
            padding: 0;
            opacity: 0;
            visibility: hidden;
            transition: 0.2s;
          }

          .fab-action-button__icon {
            display: inline-block;
            width: 56px;
            height: 56px;
            background: url("/static/img/social/action.svg#action") center no-repeat;
          }

          .fab-buttons__item {
            display: block;
            text-align: center;
            margin: 12px 0;
          }

          .fab-buttons__link {
            display: inline-block;
            width: 40px;
            height: 40px;
            text-decoration: none;
            background-color: #ffffff;
            border-radius: 50%;
            box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 1px 5px 0 rgba(0, 0, 0, 0.12), 0 3px 1px -2px rgba(0, 0, 0, 0.2);
            transform: scaleY(0.5) scaleX(0.5) translateY(0px) translateX(0px);
            -moz-transition: .3s;
            -webkit-transition: .3s;
            -o-transition: .3s;
            transition: .3s;
          }

          .icon-material {
            display: inline-block;
            width: 40px;
            height: 40px;
          }

          .icon-material_fb {
            background: url("/static/img/social/fb.svg") center no-repeat;
          }

          .icon-material_gm {
            background: url("/static/img/social/gm.svg") center no-repeat;
          }

          .icon-material_li {
            background: url("/static/img/social/linkedin.svg") center no-repeat;
          }
        `}
      </style>
    </div>
  );
}

export default SocialFabNoSSR;
