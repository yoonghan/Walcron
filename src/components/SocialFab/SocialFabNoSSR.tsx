
`use strict`

import * as React from "react";
import Modal from "../Modal";
import EmailSender from "../LetterBox/EmailSender";

export interface SocialFabNoSSRProps {
}

const SocialFabNoSSR: React.SFC<SocialFabNoSSRProps> = ({}) => {
  const [isEmailShown, setEmailShown] = React.useState(false);

  function onOpenEmail() {
    setEmailShown(true);
  }

  function closeCallback() {
    setEmailShown(false);
  }

  return (
    <div className="fab">
      <div className="fab-action-button">
        <svg className="fab-action-button__icon">
          <use xlinkHref="/img/social/social-sprite.svg#action"></use>
        </svg>
      </div>
      <ul className="fab-buttons">
        <li className="fab-buttons__item">
          <a className="fab-buttons__link" target="onnew" href="//www.linkedin.com/in/han-yoong-33755361/">
            <svg className="fab-action-button__icon">
              <use xlinkHref="/img/social/social-sprite.svg#linkedin"></use>
            </svg>
          </a>
        </li>
        <li className="fab-buttons__item">
          <a className="fab-buttons__link" target="onnew" href="//github.com/yoonghan/Walcron">
            <i className="icon-material icon-material_git"></i>
          </a>
        </li>
        <li className="fab-buttons__item">
          <a className="fab-buttons__link" target="onnew" href="//stackoverflow.com/users/3893990/han">
            <svg className="fab-action-button__icon">
              <use xlinkHref="/img/social/social-sprite.svg#so"></use>
            </svg>
          </a>
        </li>
        <li className="fab-buttons__item">
          <a className="fab-buttons__link" target="onnew" href="//www.facebook.com/walcron.coorperation">
            <svg className="fab-action-button__icon">
              <use xlinkHref="/img/social/social-sprite.svg#fb"></use>
            </svg>
          </a>
        </li>
        <li className="fab-buttons__item">
          <a className="fab-buttons__link" onClick={onOpenEmail}>
            <svg className="fab-action-button__icon">
              <use xlinkHref="/img/social/social-sprite.svg#gm"></use>
            </svg>
          </a>
        </li>
      </ul>
      {
        isEmailShown && (<Modal cancelCallback={closeCallback}>
          <EmailSender writeTo={"Walcron Website"} writeFrom={""} />
          </Modal>)
      }
      <style jsx>
        {`
          a:hover {
            cursor: pointer;
          }

          .fab {
            position: fixed;
            width: 56px;
            right: 15px;
            bottom: 15px;
            margin-left: -28px;
          }

          .fab:hover .fab-buttons {
            opacity: 1;
            visibility: visible;
          }

          .fab:hover .fab-buttons__link {
            transform: scaleY(1) scaleX(1) translateY(-16px) translateX(0px);
          }

          .fab-action-button:hover + .fab-buttons .fab-buttons__link:before {
            visibility: visible;
            opacity: 1;
            transform: scale(1);
            transform-origin: right center 0;
            transition-delay: 0.3s;
          }

          .fab-action-button {
            position: absolute;
            bottom: 0;
            display: flex;
            justify-content: center;
            align-items: center;
            width: 56px;
            height: 56px;
            background-color: #000;
            border-radius: 50%;
            box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 1px 5px 0 rgba(0, 0, 0, 0.12), 0 3px 1px -2px rgba(0, 0, 0, 0.2);
          }

          .fab-buttons {
            position: absolute;
            left: 0;
            right: 0;
            bottom: 30px;
            list-style: none;
            margin: 0;
            padding: 0;
            opacity: 0;
            visibility: hidden;
            transition: 0.2s;
          }

          .fab-action-button__icon {
            display: inline-block;
            width: 40px;
            height: 40px;
          }

          .fab-buttons__item {
            display: block;
            text-align: center;
            margin: 12px 0;
          }

          .fab-buttons__link {
            display: inline-block;
            width: 40px;
            height: 40px;
            text-decoration: none;
            background-color: #ffffff;
            border-radius: 50%;
            box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 1px 5px 0 rgba(0, 0, 0, 0.12), 0 3px 1px -2px rgba(0, 0, 0, 0.2);
            transform: scaleY(0.5) scaleX(0.5) translateY(0px) translateX(0px);
            -moz-transition: .3s;
            -webkit-transition: .3s;
            -o-transition: .3s;
            transition: .3s;
          }

          .icon-material {
            display: inline-block;
            width: 40px;
            height: 40px;
          }

          .icon-material_git {
            background: url("/img/social/git.png") center no-repeat;
            background-size: 38px;
          }
        `}
      </style>
    </div>
  );
}

export default SocialFabNoSSR;
