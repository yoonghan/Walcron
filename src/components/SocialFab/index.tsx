import * as React from "react";
import NoSSR from 'react-no-ssr';

const SocialFabNoSSRComponent = React.lazy(() => import('./SocialFabNoSSR'));

const SocialFab: React.SFC<any> = (props) => {
  return (
    <NoSSR>
      <React.Suspense fallback={<React.Fragment/>}>
        <SocialFabNoSSRComponent {...props}/>
      </React.Suspense>
    </NoSSR>
  );
}

export default React.memo(SocialFab);
