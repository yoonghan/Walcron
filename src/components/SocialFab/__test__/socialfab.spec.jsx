import * as React from 'react';
import { mount } from 'enzyme';
import SocialFab from "../index";

//Increase wait time
jest.setTimeout(5000);

//Rewrite router.
jest.mock("next/router");

//Rewrite dynamic.
jest.mock('next/dynamic');

const waitNextTick = async () => {
  return new Promise(resolve => setTimeout(resolve, 2000));
}

const mountInitialComponent = async (component) => {
  const wrapper = mount(component);
  await waitNextTick();
  wrapper.update();
  return wrapper;
};

describe('SocialFab', () => {
  it('renders lazy normally', async () => {
    const wrapper = await mountInitialComponent(<SocialFab/>);
    expect(wrapper.debug()).toMatchSnapshot();
  })
});
