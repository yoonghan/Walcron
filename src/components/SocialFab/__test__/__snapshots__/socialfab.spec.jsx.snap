// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialFab renders lazy normally 1`] = `
"<Memo(SocialFab)>
  <NoSSR>
    <Suspense fallback={{...}}>
      <SocialFabNoSSR>
        <div className=\\"jsx-3835316992 fab\\">
          <div className=\\"jsx-3835316992 fab-action-button\\">
            <svg className=\\"jsx-3835316992 fab-action-button__icon\\">
              <use xlinkHref=\\"/img/social/social-sprite.svg#action\\" className=\\"jsx-3835316992\\" />
            </svg>
          </div>
          <ul className=\\"jsx-3835316992 fab-buttons\\">
            <li className=\\"jsx-3835316992 fab-buttons__item\\">
              <a target=\\"onnew\\" href=\\"//www.linkedin.com/in/han-yoong-33755361/\\" className=\\"jsx-3835316992 fab-buttons__link\\">
                <svg className=\\"jsx-3835316992 fab-action-button__icon\\">
                  <use xlinkHref=\\"/img/social/social-sprite.svg#linkedin\\" className=\\"jsx-3835316992\\" />
                </svg>
              </a>
            </li>
            <li className=\\"jsx-3835316992 fab-buttons__item\\">
              <a target=\\"onnew\\" href=\\"//github.com/yoonghan/Walcron\\" className=\\"jsx-3835316992 fab-buttons__link\\">
                <i className=\\"jsx-3835316992 icon-material icon-material_git\\" />
              </a>
            </li>
            <li className=\\"jsx-3835316992 fab-buttons__item\\">
              <a target=\\"onnew\\" href=\\"//stackoverflow.com/users/3893990/han\\" className=\\"jsx-3835316992 fab-buttons__link\\">
                <svg className=\\"jsx-3835316992 fab-action-button__icon\\">
                  <use xlinkHref=\\"/img/social/social-sprite.svg#so\\" className=\\"jsx-3835316992\\" />
                </svg>
              </a>
            </li>
            <li className=\\"jsx-3835316992 fab-buttons__item\\">
              <a target=\\"onnew\\" href=\\"//www.facebook.com/walcron.coorperation\\" className=\\"jsx-3835316992 fab-buttons__link\\">
                <svg className=\\"jsx-3835316992 fab-action-button__icon\\">
                  <use xlinkHref=\\"/img/social/social-sprite.svg#fb\\" className=\\"jsx-3835316992\\" />
                </svg>
              </a>
            </li>
            <li className=\\"jsx-3835316992 fab-buttons__item\\">
              <a onClick={[Function: onOpenEmail]} className=\\"jsx-3835316992 fab-buttons__link\\">
                <svg className=\\"jsx-3835316992 fab-action-button__icon\\">
                  <use xlinkHref=\\"/img/social/social-sprite.svg#gm\\" className=\\"jsx-3835316992\\" />
                </svg>
              </a>
            </li>
          </ul>
          <JSXStyle id=\\"3835316992\\" />
        </div>
      </SocialFabNoSSR>
    </Suspense>
  </NoSSR>
</Memo(SocialFab)>"
`;
